hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java

```diff
@@ -1024,6 +1027,98 @@ private int getWhenNoSSLDetected() {
+            = new HazelcastProperty("hazelcast.metrics.enabled");
+
+    /**
+     * Enables/disables collecting metrics for Management Center. If disabled,
+     * Management Center can't consume the metrics from this member.
+     * <p/>
+     * NOTE: This property overrides {@link MetricsManagementCenterConfig#isEnabled()}.
+     * <p/>
+     * Using {@link MetricsManagementCenterConfig#setEnabled(boolean)} and the declarative
+     * counterparts are preferred over using this property. The main purpose
+     * of making metrics collection configurable from properties too is
+     * allowing operators to configure the metrics subsystem from the outside
+     * during investigation without touching or copying the configuration
+     * potentially embedded into a signed artifact.
+     */
+    public static final HazelcastProperty METRICS_MC_ENABLED
+            = new HazelcastProperty("hazelcast.metrics.mc.enabled");
+
+    /**
+     * Sets the duration in seconds for which the collected metrics are retained
+     * and Management Center can consume them.
+     * <p/>
+     * NOTE: This property overrides {@link MetricsManagementCenterConfig#getRetentionSeconds()}.
+     * <p/>
+     * Using {@link MetricsManagementCenterConfig#setRetentionSeconds(int)} and the declarative
+     * counterparts are preferred over using this property. The main purpose
+     * of making metrics collection configurable from properties too is
+     * allowing operators to configure the metrics subsystem from the outside
+     * during investigation without touching or copying the configuration
+     * potentially embedded into a signed artifact.
+     */
+    public static final HazelcastProperty METRICS_MC_RETENTION
+            = new HazelcastProperty("hazelcast.metrics.mc.retention");
+
+    /**
+     * Enables/disables exposing metrics on JMX.
+     * <p/>
+     * NOTE: This property overrides {@link MetricsJmxConfig#isEnabled()}.
+     * <p/>
+     * Using {@link MetricsJmxConfig#setEnabled(boolean)} and the declarative
+     * counterparts are preferred over using this property. The main purpose
+     * of making metrics collection configurable from properties too is
+     * allowing operators to configure the metrics subsystem from the outside
+     * during investigation without touching or copying the configuration
+     * potentially embedded into a signed artifact.
+     */
+    public static final HazelcastProperty METRICS_JMX_ENABLED
+            = new HazelcastProperty("hazelcast.metrics.jmx.enabled");
+
+    /**
+     * Enables collecting debug metrics. Debug metrics are sent to the
+     * diagnostics only.
+     */
+    public static final HazelcastProperty METRICS_DEBUG
+            = new HazelcastProperty("hazelcast.metrics.debug.enabled");
+
+    /**
+     * Enables/disables collecting metrics from the distributed data structures.
+     * By default, collecting data structure metrics are enabled.
```

Comment:
Potential misprint: `collecting data structure metrics are enabled` should be `data structure metrics collection is enabled`.

Reply:
:+1: Replaced